domain/externalcontroller/service/service.go

```diff
@@ -62,3 +74,9 @@ func (s *Service) UpdateExternalController(
 	err := s.st.UpdateExternalController(ctx, ec, modelUUIDs)
 	return errors.Annotate(err, "updating external controller state")
 }
+
+func (s *Service) Watch() (watcher.StringsWatcher, error) {
+	return s.watcherFactory.NewUUIDsWatcher(
+		changestream.Create|changestream.Update, "external_controller",
```

Comment:
This isn't specific to this PR, but in general to the original implementation. What happens if an external_controller is deleted, the other side will still attempt to listen because it's never told to stop listening. Seems a bit weird.